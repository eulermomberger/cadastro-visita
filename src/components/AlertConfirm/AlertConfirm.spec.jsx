import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';

import { AlertConfirm } from '.';

describe('AlertConfirm component', () => {
  it('Should appear when the isOpen prop is equal to true', () => {
    const testRenderer = renderer.create(
      <AlertConfirm
        isOpen
        title='Alerta de confirmação'
        description='Descrição para o alerta de confirmação'
        onCancel={() => {}}
        onConfirm={() => {}}
      />
    );

    try {
      expect(testRenderer.toJSON()).toMatchSnapshot();
    } finally {
      testRenderer.unmount();
    }
  });

  it('Should desappear when the isOpen prop is equal to false', () => {
    const testRenderer = renderer.create(
      <AlertConfirm
        isOpen={false}
        title='Alerta de confirmação'
        description='Descrição para o alerta de confirmação'
        onCancel={() => {}}
        onConfirm={() => {}}
      />
    );

    try {
      expect(testRenderer.toJSON()).toMatchSnapshot();
    } finally {
      testRenderer.unmount();
    }
  });

  it('Should call the functions prop when clicked', () => {
    const mockCancelFunc = jest.fn();
    const mockConfirmFunc = jest.fn();

    const { getByText } = render(
      <AlertConfirm
        isOpen={true}
        title='Alerta de confirmação'
        description='Descrição para o alerta de confirmação'
        onCancel={mockCancelFunc}
        onConfirm={mockConfirmFunc}
      />
    );

    fireEvent.click(getByText(/Cancelar/i));
    expect(mockCancelFunc).toHaveBeenCalled();

    fireEvent.click(getByText(/Confirmar/i));
    expect(mockConfirmFunc).toHaveBeenCalled();
  });
});
