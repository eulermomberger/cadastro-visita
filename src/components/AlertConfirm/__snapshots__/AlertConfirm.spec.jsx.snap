// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertConfirm component Should appear when the isOpen prop is equal to true 1`] = `
<div
  className="backdrop-alert"
>
  <div
    className="alert-container"
  >
    <div
      className="alert-header"
    >
      <h2
        className="alert-title"
      >
        Alerta de confirmação
      </h2>
    </div>
    <div
      className="alert-body"
    >
      <span>
        Descrição para o alerta de confirmação
      </span>
    </div>
    <div
      className="alert-footer"
    >
      <button
        className="alert-button alert-button-cancel"
        onClick={[Function]}
      >
        Cancelar
      </button>
      <button
        className="alert-button alert-button-confirm"
        onClick={[Function]}
      >
        Confirmar
      </button>
    </div>
  </div>
</div>
`;

exports[`AlertConfirm component Should desappear when the isOpen prop is equal to false 1`] = `null`;
